conf/config.go

```diff
@@ -190,3 +216,102 @@ func UnsetEnv(protect *testing.T, key string) error {
 	return err
 
 }
+
+/***********************************************************************************************
+
+    CODE BELOW THIS BLOCK IS EXPERIMENTAL AND ONLY BEING USED INTERNALLY BY CONF PACKAGE
+
+***********************************************************************************************/
+
+// Gopath function exposes the gopath of the application while keeping it immutable. Golang does
+// not allow const to be strings, and a var would make it mutable if made public. This could be
+// something the config / viper struct keeps track. For now it's separate.
+func Gopath() string {
+	return envVars.gopath
+}
+
+/*
+Checkout function takes a reference to a struct or a slice of string. It will traverse both
+data structures and look up key value pairs in the conf / viper struct by the name of the field
+for structs and string values for the slice. The function works with pointers so no value is
+returned. An error is returned if the wrong data structure is provided.
+*/
+func Checkout(list interface{}) error {
+
+	// Check if the data type provided is supported
+	switch list := list.(type) {
+	// If it's a slice of strings
+	case []string:
+
+		for n, str := range list {
+			if val, exists := LookupEnv(str); exists {
+				list[n] = val
+			}
+		}
+
+		return nil
+	// Checking the rest of data types through reflection
+	default:
+		// Get the concrete value from the interface
+		check := reflect.ValueOf(list)
+		// Is it a pointer?
+		if check.Kind() == reflect.Ptr {
+
+			// Dereference the pointer
+			el := check.Elem()
+
+			// Is the data type a struct?
+			if el.Kind() == reflect.Struct {
+
+				// Get ready to walk the fields of the structs
+				num := el.NumField() // Number of fields in struct
+				elType := el.Type()  // Used for tags
+
+				for i := 0; i < num; i++ {
+					field := el.Field(i)
+					// Ensure the field is "exportable" and is a type string
+					if field.CanInterface() && field.IsValid() && field.Type().Name() == "string" {
```

Comment:
Might be something later down the road - but should we add support for other data types? 

Here's a reference on how the json.Unmarshal is achieving the similar functionality with reflection: https://golang.org/src/encoding/json/decode.go?s=4081:4129#L784

Reply:
I think so!

Perhaps we start with the two we have now, and expand if needed during ALR implementation?

Reply:
Yup that works for me. My thinking with expanded types is to catch the situation where we say that we expect a numeric value, but it turns out that the value is set to some non-numeric input. I'm thinking we would want to catch this as soon as possible, similar to how JSON unmarshal would fail in the similar situation.